server/src/main/java/org/elasticsearch/cluster/ShardHeapUsageSupplier.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.cluster;
+
+import org.elasticsearch.action.ActionListener;
+
+import java.util.Map;
+
+public interface ShardHeapUsageSupplier {
```

Comment:
nit: this isn't really a supplier, it's not creating them exactly, it's collecting them from the cluster, so maybe "Collector" suffix, and collect method?

Reply:
Addressed in 7b5bf95d37849d3c93cf69b6594c4629057f7130